server/neptune/gateway/event/push_event_handler.go

```diff
@@ -104,6 +105,12 @@ func (p *PushHandler) handle(ctx context.Context, event Push) error {
 
 func (p *PushHandler) startWorkflow(ctx context.Context, event Push, rootCfg *valid.MergedProjectCfg) (client.WorkflowRun, error) {
 	options := client.StartWorkflowOptions{TaskQueue: workflows.DeployTaskQueue}
+
+	var tfVersion string
+	if rootCfg.TerraformVersion != nil {
+		tfVersion = rootCfg.TerraformVersion.String()
+	}
```

Comment:
Should we fail if `tfVersion` is empty?

Reply:
this version is parsed from the atlantis.yaml file which does not enforce specifying the terraform versions for the root. So, if the version is not specified, we use the default tf version configured during startup. This is how atlantis currently handles terraform version afaik!